### PR TITLE
admin_orders_partial

### PR DESCRIPTION
**Pull request by MaciejO98**

### CODE REVIEW

File: app/models/material.rb [hunk 1]
```diff
@@ -211,4 +216,13 @@ def has_credits?(receiver, amount)
 
     receiver.profile.credits >= (amount * cost)
   end
+
+  def create_order_for_dispensary
+    return unless self.owner.profile.dispensary?
```
Comment (Shalashtein) [inline app/models/material.rb:221]: There's no method called `dispensary?` use `.role != 'dispensary` instead

File: app/models/material.rb [hunk 2]
```diff
@@ -211,4 +216,13 @@ def has_credits?(receiver, amount)
...
+    return if self.amount > DISPENSARY_LOW_AMOUNT
+    warehouse_material = Material.includes(owner: :profile).includes(:material_type).where(owner: {profiles: {role: :warehouse}}, material_type: {id: self.material_type_id}).first
```
Comment (Shalashtein) [inline app/models/material.rb:223]: Multiple Material instances might have the same type, perhaps if we just include the id of the material that's lo on stock somewhere

**Review by Shalashtein — CHANGES_REQUESTED** [review]

File: app/models/material.rb (hunk 1 above)
Comment (wolak88) [reply app/models/material.rb:221]: @Shalashtein 
In rails creating an enum creates scope on model, so you can do `Profile.dispensary` and you will get only profiles that are dispensaries also you get `dispensary?` and some other methods

File: app/models/material.rb (hunk 2 above)
Comment (wolak88) [reply app/models/material.rb:223]: Yeah, I think we should create `MaterialRequest` class where we will store material type, amount and user(dispensary) which is making the request. I don't know if it should be connected to particular warehouse or shouldn't

**Review by Shalashtein — APPROVED** [review]

**Pull request merged**